WordPress/Classes/ViewRelated/Me/All Domains/Views/AllDomainsListEmptyView.swift

```diff
@@ -0,0 +1,101 @@
+import UIKit
+import WordPressUI
+
+final class AllDomainsListEmptyView: UIView {
+
+    typealias ViewModel = AllDomainsListEmptyStateViewModel
+
+    private enum Appearance {
+        static let labelsSpacing: CGFloat = Length.Padding.single
+        static let labelsButtonSpacing: CGFloat = Length.Padding.medium
+        static let titleLabelFont: UIFont = WPStyleGuide.fontForTextStyle(.title2, fontWeight: .bold)
+        static let descriptionLabelFont: UIFont = WPStyleGuide.fontForTextStyle(.callout, fontWeight: .regular)
+        static let buttonLabelFont: UIFont = WPStyleGuide.fontForTextStyle(.body, fontWeight: .regular)
+        static let titleLabelColor: UIColor? = UIColor.DS.Foreground.secondary
+        static let descriptionLabelColor: UIColor? = UIColor.DS.Foreground.secondary
+    }
+
+    // MARK: - Actions
+
+    private var buttonAction: (() -> Void)?
+
+    // MARK: - Views
+
+    private let titleLabel: UILabel = {
+        let label = UILabel()
+        label.textAlignment = .center
+        label.font = Appearance.titleLabelFont
+        label.textColor = Appearance.titleLabelColor
+        label.numberOfLines = 2
+        label.adjustsFontForContentSizeCategory = true
+        return label
+    }()
+
+    private let descriptionLabel: UILabel = {
+        let label = UILabel()
+        label.textAlignment = .center
+        label.font = Appearance.descriptionLabelFont
+        label.textColor = Appearance.descriptionLabelColor
+        label.numberOfLines = 0
+        label.adjustsFontForContentSizeCategory = true
+        return label
+    }()
+
+    private let button: UIButton = {
+        let button = FancyButton()
+        button.titleLabel?.font = Appearance.buttonLabelFont
+        button.isPrimary = true
+        return button
+    }()
+
+    // MARK: - Init
+
+    init(viewModel: ViewModel? = nil) {
+        super.init(frame: .zero)
+        self.render(with: viewModel)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Rendering
+
+    private func render(with viewModel: ViewModel?) {
```

Comment:
By no means this is a change request but I reckon with SwiftUI this file would be quite shorter than what we get with UIKit. Was there a reason to prefer UIKit over it?

Reply:
Hey @alpavanoglu, my first thought was to implement it in SwiftUI but I changed my mind for these reasons:
1. I couldn't find a SwiftUI alternative for that "FancyButton" component ( the primary button ). And I thought recreating that component and making sure it is identical to "FancyButton" is not within the scope of the project.
2. I wanted to try the UIKit #Preview feature.

That being said, I'm open to revisit this later and rewrite it to SwiftUI.